src/ducks/balance/components/AccountRowLoading.jsx

```diff
@@ -7,58 +7,77 @@ import KonnectorIcon from 'ducks/balance/components/KonnectorIcon'
 import styles from 'ducks/balance/components/AccountRow.styl'
 import stylesLoading from 'ducks/balance/components/AccountRowLoading.styl'
 import cx from 'classnames'
+import { Intents } from 'cozy-interapp'
 
-const AccountRowLoading = ({ t, institutionSlug, status }) => {
-  const isErrored = status === 'errored'
-  const liProps = isErrored ? { onClick: () => alert('ok') } : {}
-  return (
-    <li
-      className={cx(styles.AccountRow, {
-        [stylesLoading.pointer]: isErrored
-      })}
-      {...liProps}
-    >
-      <div className={styles.AccountRow__column}>
-        <div className={styles.AccountRow__logo}>
-          {institutionSlug && (
-            <KonnectorIcon
-              slug={institutionSlug}
-              className={styles.KonnectorIcon}
-            />
-          )}
-        </div>
-        <div className={styles.AccountRow__labelUpdatedAtWrapper}>
-          <div className={styles.AccountRow__label}>
-            {isErrored
-              ? t('Balance.importing_accounts_error')
-              : t('Balance.importing_accounts')}
-          </div>
-          <div className={styles.AccountRow__updatedAt}>
-            {isErrored ? (
-              <>
-                <Icon size="12" icon="warning" />
-                <span className={stylesLoading.error}>
-                  {t('Balance.error')}
-                </span>
-              </>
-            ) : (
-              <>
-                <Icon
-                  size="12"
-                  icon="spinner"
-                  color="var(--primaryColor)"
-                  spin
-                />
-                <span className={stylesLoading.InProgress}>
-                  {t('Balance.in_progress')}
-                </span>
-              </>
+class AccountRowLoading extends Component {
+  static contextTypes = {
+    client: PropTypes.object.isRequired
+  }
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.intents = new Intents({ client: this.context.client })
```

Comment:
Nit: you can wrap `AccountRowLoading` in `withClient` cozy-client's HOC to avoid hitting the context directly.

Reply:
what is the advantage?

Reply:
You get the client directly in the props and you don't have to specify the `contextTypes`. But I see that here you didn't specified the `contextTypes`. This is weird that it works since as far as I know we use the legacy context API and [the doc](https://reactjs.org/docs/legacy-context.html#how-to-use-context) says : « If contextTypes is not defined, then context will be an empty object. »